example/spec/parallel/SyncGranuleTask/SyncZeroByteFileSpec.js

```diff
@@ -0,0 +1,162 @@
+        prefix,
+        {
+          duplicateHandling: 'version',
+          process: 'modis',
+        }
+      );
+
+      // Create the S3 provider
+      provider = await createProvider(prefix, { host: fakeS3ProviderBucket });
+
+      granuleId = randomId('granule-id-');
+
+      const FunctionName = `${prefix}-SyncGranule`;
+      const functionConfig = await lambda().getFunctionConfiguration({
+        FunctionName,
+      }).promise();
+
+      const Payload = JSON.stringify({
+        cma: {
+          task_config: {
+            buckets: '{$.meta.buckets}',
+            provider: '{$.meta.provider}',
+            collection: '{$.meta.collection}',
+            stack: '{$.meta.stack}',
+            downloadBucket: '{$.cumulus_meta.system_bucket}',
+            duplicateHandling: '{$.meta.collection.duplicateHandling}',
+            cumulus_message: {
+              input: '{$.payload}',
+              outputs: [
+                {
+                  source: '{$}',
+                  destination: '{$.payload}',
+                },
+              ],
+            },
+          },
+          event: {
+            cumulus_meta: {
+              system_bucket: config.bucket,
+            },
+            meta: {
+              buckets: config.buckets,
+              collection,
+              provider,
+              stack: config.stackName,
+            },
+            payload: {
+              granules: [
+                {
+                  granuleId,
+                  dataType: collection.name,
+                  version: collection.version,
+                  files: [
+                    {
+                      bucket: fakeS3ProviderBucket,
+                      name: filename,
+                      path: '',
+                      size: 0,
+                    },
+                  ],
+                },
+              ],
+            },
+          },
+        },
+      });
+
+      syncGranuleOutput = await pTimeout(
+        lambda().invoke({ FunctionName, Payload }).promise(),
+        (functionConfig.Timeout + 10) * 1000
+      );
+    } catch (error) {
+      beforeAllFailed = true;
+      throw error;
+    }
+  });
+
+  it('succeeds', async () => {
+    if (beforeAllFailed) fail('beforeAll() failed');
+    else {
+      expect(syncGranuleOutput.FunctionError).toBe(undefined);
+
+      const parsedPayload = JSON.parse(syncGranuleOutput.Payload);
+      const fullTaskOutput = await pullStepFunctionEvent(parsedPayload);
+
+      expect(fullTaskOutput.errorType).toBe(undefined);
+      expect(fullTaskOutput.errorMessage).toBe(undefined);
+
+      // Confirm that file was synced
+      const syncedFile = parsedPayload.payload.granules[0].files[0];
+      parsedFileUrl = S3.parseS3Uri(syncedFile.filename);
+      expect(
+        await S3.s3ObjectExists({
+          Bucket: parsedFileUrl.Bucket,
+          Key: parsedFileUrl.Key,
+        })
+      ).toBeTrue();
+    }
+  });
+
+  afterAll(async () => {
+    await pAll(
```

Comment:
Also curious: I'm not familiar with `p-all`. It looks like the benefit here is that we can try to resolve all of the promises with `{ stopOnError: false}` rather than stopping at the first rejection. This way we can get an aggregate of all the results and have better error logging. Is that about right? Is there another reason it's used here in place of Promise.all?

Reply:
Yeah, we want to run as much of the cleanup as possible, and `p-all` with `{ stopOnError: false }` will run all of the promises even if any one of them fails, whereas `Promise.all` would stop as soon as any one of the promises failed.